apps/api/src/routers/saml.py

```diff
@@ -81,6 +83,13 @@ async def _get_saml_auth(req: Request) -> OneLogin_Saml2_Auth:
     return OneLogin_Saml2_Auth(request_data, old_settings=settings)
 
 
+async def _insert_native_record(user: NativeUser) -> None:
```

Comment:
Nitpick: "insert" isn't always accurate here since when the record already exists, we're just updating it. Perhaps consider `_update_last_login` or `_save_login_time` or alike.